ui/components/app/contact-list/contact-list.component.js

```diff
@@ -50,34 +51,36 @@ export default class ContactList extends PureComponent {
   }
 
   renderAddressBook() {
-    const contacts = this.props.searchForContacts();
+    const unsortedContactsByLetter = this.props
+      .searchForContacts()
+      .reduce((obj, contact) => {
+        const firstLetter = contact.name[0].toUpperCase();
+        return {
+          ...obj,
+          [firstLetter]: [...(obj[firstLetter] || []), contact],
+        };
+      }, {});
 
-    const contactGroups = contacts.reduce((acc, contact) => {
-      const firstLetter = contact.name.slice(0, 1).toUpperCase();
-      acc[firstLetter] = acc[firstLetter] || [];
-      const bucket = acc[firstLetter];
-      bucket.push(contact);
-      return acc;
-    }, {});
+    const letters = Object.keys(unsortedContactsByLetter).sort();
 
-    return Object.entries(contactGroups)
-      .sort(([letter1], [letter2]) => {
-        if (letter1 > letter2) {
-          return 1;
-        } else if (letter1 === letter2) {
-          return 0;
-        }
-        return -1;
-      })
-      .map(([letter, groupItems]) => (
-        <RecipientGroup
-          key={`${letter}-contract-group`}
-          label={letter}
-          items={groupItems}
-          onSelect={this.props.selectRecipient}
-          selectedAddress={this.props.selectedAddress}
-        />
-      ));
+    const sortedContactGroups = letters.map((letter) => {
+      return [
+        letter,
+        sortBy(unsortedContactsByLetter[letter], (contact) => {
+          return contact.name.toLowerCase();
```

Comment:
Note that each contact is sorted case-insensitively. Personally I find this less confusing as when I look through I list that I know is sorted, I don't think about case. But I'm happy to change this if we think otherwise.